src/posix/handlers/getdents.hpp

```diff
@@ -20,16 +21,16 @@ inline off64_t round(off64_t bytes, bool is_getdents64) {
 // TODO: too similar to capio_read, refactoring needed
 inline int getdents_handler_impl(long arg0, long arg1, long arg2, long *result, bool is64bit) {
     auto fd      = static_cast<int>(arg0);
-    auto *buffer = reinterpret_cast<struct linux_dirent *>(arg1);
+    auto *buffer = reinterpret_cast<struct dirent *>(arg1);
```

Comment:
I stille do not understand why here you use the `struct dirent` from `glibc` and on the server you use the custom `linux_dirent64` structure. If they are equal, use the standard `dirent` in both places. If they are different ,why do you use this one here?

Reply:
Resolved using linuc_dirent64, and by moving the datastructure.hpp to capio/common